src/VirtualDevice.ts

```diff
@@ -248,46 +249,62 @@ export class VirtualDevice {
 
         const url = URL.parse(this.baseURL);
 
-        return new Promise<IVirtualDeviceResponse | any>((resolve, reject) => {
-            const callback = (response: IncomingMessage) => {
-                let data = "";
-
-                response.on("data", (chunk) => {
-                    data += chunk;
-                });
+        const responsePromise = () => {
+            return new Promise<IVirtualDeviceResponse | any>((resolve, reject) => {
+                const callback = (response: IncomingMessage) => {
+                    let data = "";
+
+                    response.on("data", (chunk) => {
+                        data += chunk;
+                    });
+
+                    response.on("end", () => {
+                        if (response.statusCode === 200) {
+                            const result = this.handleBatchResponse(data as string);
+                            if ((result as IVirtualDeviceError).error) {
+                                reject(result);
+                                return;
+                            }
 
-                response.on("end", () => {
-                    if (response.statusCode === 200) {
-                        const result = this.handleBatchResponse(data as string);
-                        if ((result as IVirtualDeviceError).error) {
-                            reject(result);
-                            return;
+                            resolve(result);
+                        } else {
+                            reject(data);
                         }
-
-                        resolve(result);
-                    } else {
-                        reject(data);
-                    }
+                    });
+                };
+
+                const requestOptions = {
+                    headers: {
+                        "Content-Type": "application/json",
+                    },
+                    host: url.hostname,
+                    method: "GET",
+                    path,
+                    port: this.httpInterfacePort(url),
+                };
+
+                const request = this.httpInterface(url).request(requestOptions, callback);
+                request.on("socket", (socket: any) => {
+                    socket.setTimeout(2000);
```

Comment:
this timeouts are ok, but we should set an environment variable to be able to change them if we want while keeping these values for default (this will also help to avoid long tests). 
Another option is to add them to the class constructor